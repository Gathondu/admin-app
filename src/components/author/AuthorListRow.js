import React from 'react';
import PropTypes from 'prop-types';
import {Link} from "react-router-dom";


const AuthorListRow = ({author, deleteAuthor}) => {
    return (
        <tr>
            <td><i className="fa fa-trash" onClick={() => deleteAuthor(author.id)}/></td>
            <td><Link to={`/author/${author.id}`}>{author.firstName + ' ' + author.lastName}</Link></td>
        </tr>
    );
};

AuthorListRow.propTypes = {
    author: PropTypes.object.isRequired,
    deleteAuthor: PropTypes.func.isRequired
};
export default AuthorListRow;
