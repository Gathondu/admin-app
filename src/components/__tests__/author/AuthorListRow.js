import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AuthorListRow from "../../author/AuthorListRow";

describe('AuthorListRow Component',()=>{
    const setup = () => {
        const props = {
            author: {},
            deleteAuthor: jest.fn()
        };
        return {
            props,
            wrapper: shallow(<AuthorListRow {...props}/>)
        }
    };
    it('does not regress', ()=>{
        const {wrapper} = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should delete author correctly', () => {
        const {wrapper, props} = setup();
        wrapper.find('i').simulate('click');
        expect(props.deleteAuthor).toHaveBeenCalled()
    });
});