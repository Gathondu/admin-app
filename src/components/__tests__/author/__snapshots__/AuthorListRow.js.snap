// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorListRow Component does not regress 1`] = `
<tr>
  <td>
    <i
      className="fa fa-trash"
      onClick={[Function]}
    />
  </td>
  <td>
    <Link
      replace={false}
      to="/author/undefined"
    >
      undefined undefined
    </Link>
  </td>
</tr>
`;
